File: python/basic_grammar/calculation/numeric_operation/addition.py
from typing import Union

natural_number_addition_case = 1 + 2
decimal_addition_case = 2.1 + 4.2
negative_number_addition_case = -8 + 4
string_addition_case = "Hello, " + "world!"
array_addition_case: list[str] = ["apple", "banana"] + ["orange"]
dictionary_addition_case: dict[str, Union[str, int]] = {'name': "Tom", 'age': 34} | {'breakfirst': "apple"}

print(natural_number_addition_case)
print(decimal_addition_case)
print(negative_number_addition_case)
print(string_addition_case)
print(array_addition_case)
print(dictionary_addition_case)

# 単にコンソールへ出力するという場合は以下のような書き方でも連結できる
# print("Hello,", "world!")

File: c++/basic_grammar/calculation/numeric_operation/addition.cpp
#include <iostream>
using namespace std;

int main() {
  cout << 1 + 2 << endl;
  cout << 2.1 + 4.2 << endl;
  cout << -8 + 4 << endl;

  // 1.半角スペースで結合する例
  cout << "Hello, " "world!" << endl;

  // 2.std::string型で変数を宣言して連結する例
  std::string hello_world = "Hello, ";
  hello_world += "world!";
  cout << hello_world << endl;
  
  // 以下のコードはエラーが発生する
  // cout << "Hello, " + "world!" << endl;
}